magda-io/magda
# Auth on record summary

### What this PR does

Fixes #2372 

This adds authorization to the /records/summary/<recordid> API, to make it work pretty much the same way that the `/records/<recordid>` API does - i.e. if the user isn't allowed to see a record, it returns 404. The work around this actually pretty much all tests because it uses much of the same code to talk to the database and therefore works already.

In addition, I fixed some bugs:
- You can now update `authnRecordPolicyId` from `PUT /records/<id>` (before this returned 200 but didn't change the policy, or return its value)
- Made it so records with no aspects at all still show up (previously they were _always_ marked as not authorized.

#### To Test
There are two records in https://issues-2372-record-summary-auth.dev.magda.io/api/v0/registry/records - one is called `record-public` and one is called `record-private`. `record-public` is set to public, so `/records/summary/record-public` should always return 200. `record-private` is set to `owner_orgunit`, so `records/summary/record-private` should only return 200 if you're currently logged in and have a unit matching the policy, otherwise it should return 404.

### Checklist
